tests/test_access_points_list/test_access_points_list.cpp

```diff
@@ -0,0 +1,580 @@
+
+    const wifi_ap_record_t exp_arr_of_aps[2] = {
+        {
+            .ssid = { 'a', 'b', 'c', '\0' },
+        },
+        {
+            .ssid = { 'a', 'b', 'c', '\0' },
+        },
+    };
+    ASSERT_EQ(exp_arr_of_aps[0], arr_of_aps[0]);
+}
+
+TEST_F(TestAccessPointsList, ap_list_reorder_2_both_non_empty_not_equal) // NOLINT
+{
+    wifi_ap_record_t arr_of_aps[2] = {
+        {
+            .ssid = { 'a', 'b', 'c', '\0' },
+        },
+        {
+            .ssid = { 'q', 'w', 'e', '\0' },
+        },
+    };
+    ASSERT_EQ(2, ap_list_reorder(arr_of_aps, sizeof(arr_of_aps) / sizeof(arr_of_aps[0])));
+
+    const wifi_ap_record_t exp_arr_of_aps[2] = {
+        {
+            .ssid = { 'a', 'b', 'c', '\0' },
+        },
+        {
+            .ssid = { 'q', 'w', 'e', '\0' },
+        },
+    };
+    ASSERT_EQ(exp_arr_of_aps[0], arr_of_aps[0]);
+}
+
+TEST_F(TestAccessPointsList, ap_list_reorder_complex) // NOLINT
+{
+    wifi_ap_record_t arr_of_aps[] = {
+        { 0 },
+        {
+            .ssid = { 'a', 'b', 'c', '\0' },
+        },
+        { 0 },
+        {
+            .ssid = { 'q', 'w', 'e', '\0' },
+        },
+        { 0 },
+    };
+    ASSERT_EQ(2, ap_list_reorder(arr_of_aps, sizeof(arr_of_aps) / sizeof(arr_of_aps[0])));
+
+    const wifi_ap_record_t exp_arr_of_aps[2] = {
+        {
+            .ssid = { 'a', 'b', 'c', '\0' },
+        },
+        {
+            .ssid = { 'q', 'w', 'e', '\0' },
+        },
+    };
+    ASSERT_EQ(exp_arr_of_aps[0], arr_of_aps[0]);
+}
+
+TEST_F(TestAccessPointsList, ap_list_filter_unique_empty) // NOLINT
+{
+    ASSERT_EQ(0, ap_list_filter_unique(nullptr, 0));
+}
+
+TEST_F(TestAccessPointsList, ap_list_filter_unique_complex) // NOLINT
+{
+    wifi_ap_record_t arr_of_aps[] = {
+        {
+            .ssid = { '\0' },
+        },
+        {
+            .ssid     = { 'a', 'b', 'c', '\0' },
+            .rssi     = -10,
+            .authmode = WIFI_AUTH_WPA2_PSK,
+        },
+        {
+            .ssid = { '\0' },
+        },
+        {
+            .ssid     = { 'q', 'w', 'e', '\0' },
+            .rssi     = -20,
+            .authmode = WIFI_AUTH_WPA2_PSK,
+        },
+        {
+            .ssid = { '\0' },
+        },
+        {
+            .ssid     = { 'a', 'b', 'c', '\0' },
+            .rssi     = -10,
+            .authmode = WIFI_AUTH_WPA3_PSK,
+        },
+        {
+            .ssid     = { 'a', 'b', 'c', '\0' },
+            .rssi     = -40,
+            .authmode = WIFI_AUTH_WPA2_PSK,
+        },
+        {
+            .ssid     = { 'q', 'w', 'e', '\0' },
+            .rssi     = -30,
+            .authmode = WIFI_AUTH_WPA2_PSK,
+        },
+        {
+            .ssid     = { 'a', 'b', 'c', '\0' },
+            .rssi     = -9,
+            .authmode = WIFI_AUTH_WPA2_PSK,
+        },
+    };
+    ASSERT_EQ(3, ap_list_filter_unique(arr_of_aps, sizeof(arr_of_aps) / sizeof(arr_of_aps[0])));
```

Comment:
`2 * sizeof(arr_of_aps) / sizeof(arr_of_aps[0])`?

Reply:
I don't get it, why "2 *" is needed?
`sizeof(arr_of_aps) / sizeof(arr_of_aps[0])` gives the number of elements in the array and ap_list_filter_unique requires actual number of elements in the array.

Reply:
I'm seeing 2 unique SSIDs in the list, "abc" and "qwe". I missed the first `3` in parameter list, so the original idea would be
`ASSERT_EQ(2, ap_list_filter_unique(arr_of_aps, sizeof(arr_of_aps) / sizeof(arr_of_aps[0])));`

My point is, it looks like we should test for 2 unique SSID after filtering here.

Reply:
"3" is correct result for this filtering because it compares not only SSID, but "authmode" also.
So, the third element is the sixth one in the original array (SSID=abc, authmode=WPA3)

Reply:
Ok, thanks for the clarification